File: api/nb-weather-ol/src/main/java/com/example/nb_weather_ol/model/WeatherQueries.java
package com.example.nb_weather_ol.model;

import jakarta.persistence.*;

@Entity
@Table(name = "weather_queries")
public class WeatherQueries {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String query;

    @Column(nullable = false)
    private String userId;

    // Getters and setters
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getQuery() {
        return query;

    }

    public String getUserId() {
        return userId;

    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public void setQuery(String query) {
        this.query = query;
    }

    @Override
    public String toString() {
        return "WeatherQuery{" +
                "id=" + id +
                ", query='" + query + '\'' +
                '}';
    }

}

File: backend-kotlin/src/main/kotlin/com/nbweatherproject/backend/model/WeatherQueries.kt
package com.nbweatherproject.backend

import jakarta.persistence.*

@Entity
@Table(name = "weather_queries")
data class WeatherQueries(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,

    @Column(name="query", nullable = false)
    var query: String = "",

    @Column(name="userId", nullable = false)
    var userId: String = ""
) {
    // No need for explicit getters and setters in Kotlin

    override fun toString(): String {
        return "WeatherQuery(id=$id, query='$query')"
    }
}
